### ISSUE TITLE
Ability to Share Plan Outputs Between Runs

### ISSUE THREAD

**Issue by theMagicalKarp** [dflook/terraform-github-actions#290]
### Suggestion

It would be nice if the plan output file could be shared between runs (for `terraform-apply` and `terraform-plan`), such that a plan can be staged and reviewed, then the apply can be ran with the exact plan binary. This avoids situations where the apply step could accidentally include un-reviewed changes.

My current process looks like this, and is encapsulated in a single workflow file.
1. Generate plan (and Review plan)
2. Approve plan _(via https://docs.github.com/en/actions/managing-workflow-runs/reviewing-deployments)_

**Comment by dflook** [comment]
You can certainly implement a workflow like this using `dflook/terraform-plan` and `dflook/terraform-apply`, without using the binary plan file. 

I can see the appeal of wanting to apply the exact plan that has been reviewed, but such a plan can be expected to include secrets and there isn't really any secure place to store them in GitHub. It also becomes invalid as soon as any other change is applied, which is a problem in repos that have frequent PRs.

Instead, `dflook/terraform-apply` generates a new plan and compares it to the one generated by `dflook/terraform-plan`. If they are functionally the same, i.e. changing the same resources in the same way, then the new plan is applied. This is assuming your workflow runs `dflook/terraform-apply` only after the PR has been reviewed/merged.

There are good reasons to want to apply the binary plan file though, a big one being speed - it can take a long time to generate the plan, so you may want to avoid doing it twice.

I will be working on a way to share the binary plan file in the future. Applying it will probably be through a separate action.

**Comment by RobertSheehyAtFanatics** [comment]
Thanks for the timely response and support!

That's good info, and makes sense since. In my particular scenario I do have a secure location to share the plan between jobs, which is bespoke to me.

I understand why it wouldn't make sense to the majority of folks though.

> I will be working on a way to share the binary plan file in the future. 

Thanks!

**Comment by theMagicalKarp** [comment]
Thanks for the timely response and support!

That's good info, and makes sense since. In my particular scenario I do have a secure location to share the plan between jobs, which is bespoke to me.

I understand why it wouldn't make sense to the majority of folks though.

> I will be working on a way to share the binary plan file in the future.

Thanks

**Comment by enobrev** [comment]
@dflook I wonder if a reasonable solution might be to allow a "plan_out" variable on the terraform-apply action that we can set, so we could potentially handle managing the storage and retrieval of the previously generated "plan.out" file ourselves (and worry about the security of that file ourselves), and the action would simply be responsible for using the given plan in cases where it's been provided.

**Comment by enobrev** [comment]
It seems my suggestion is already covered by #195

**Comment by dflook** [comment]
This was added in v1.40.0. There is a `plan_path` input for dflook/terraform-apply and a matching `plan_path` output for dflook/terraform-plan.

**Comment by enobrev** [comment]
Works great!  Thanks for making it happen!
